### ISSUE TITLE
autoload not working

### ISSUE THREAD

**Issue by kodikom** [wodby/docker4drupal#207]
http://drupal.docker.localhost:8000

Warning: require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17

Fatal error: require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17

### Codebase
Drupal composer template
  
### Host OS 
macOS High Sierra

### Docker info output

Containers: 13
 Running: 6
 Paused: 0
 Stopped: 7
Images: 54
Server Version: 17.09.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 455
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.49-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.54GiB
Name: moby
ID: 5P4C:36Q5:MJCZ:FMJ7:WOZC:E5MO:CLMX:X557:TVFS:HKXY:RMM5:DKCX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 70
 Goroutines: 72
 System Time: 2017-11-08T14:01:57.780427814Z
 EventsListeners: 2
No Proxy: *.local, 169.254/16
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

### Docker compose file
```yml
# Content of your docker-compose.yml file. Make sure you remove all sensible information you might have there.
```

version: "2"

services:
  mariadb:
    image: wodby/mariadb:10.1-2.3.5
    environment:
      MYSQL_ROOT_PASSWORD: password
      MYSQL_DATABASE: drupal
      MYSQL_USER: drupal
      MYSQL_PASSWORD: drupal
#    volumes:
#      - ./mariadb-init:/docker-entrypoint-initdb.d # Place init .sql file(s) here.
#      - /path/to/mariadb/data/on/host:/var/lib/mysql # I want to manage volumes manually.

  php:
# 1. Images with vanilla Drupal – wodby/drupal:[DRUPAL_VERSION]-[PHP_VERSION]-[STABILITY_TAG].
#    image: wodby/drupal:8-7.1-3.0.0
#    image: wodby/drupal:8-7.0-3.0.0
#    image: wodby/drupal:7-7.1-3.0.0
#    image: wodby/drupal:7-7.0-3.0.0
#    image: wodby/drupal:7-5.6-3.0.0
#    image: wodby/drupal:6-5.6-3.0.0
#    image: wodby/drupal:6-5.3-3.0.0
# 2. Images without Drupal – wodby/drupal-php:[PHP_VERSION]-[STABILITY_TAG].
    image: wodby/drupal-php:7.1-3.0.0
#    image: wodby/drupal-php:7.0-3.0.0
#    image: wodby/drupal-php:5.6-3.0.0
#    image: wodby/drupal-php:5.3-3.0.0
    environment:
      PHP_SENDMAIL_PATH: /usr/sbin/sendmail -t -i -S mailhog:1025
      PHP_FPM_CLEAR_ENV: "no"
      DB_HOST: mariadb
      DB_USER: drupal
      DB_PASSWORD: drupal
      DB_NAME: drupal
      DB_DRIVER: mysql
#      PHP_XDEBUG: 1
#      PHP_XDEBUG_DEFAULT_ENABLE: 1
#      PHP_XDEBUG_REMOTE_CONNECT_BACK: 0
#      PHP_XDEBUG_REMOTE_HOST: "10.254.254.254"
#      PHP_XDEBUG_PROFILER_OUTPUT_DIR: /mnt/files/xdebug/profiler
#      PHP_XDEBUG_TRACE_OUTPUT_DIR: /mnt/files/xdebug/traces
    volumes:
      - codebase:/var/www/html
## Options for macOS users (https://docker4drupal.readthedocs.io/en/latest/macos)
#      - codebase:/var/www/html:cached # User-guided caching
#      - docker-sync:/var/www/html # Docker-sync
## For Xdebug profiler files
#      - files:/mnt/files

  nginx:
# wodby/drupal-nginx:[DRUPAL_VERSION]-[NGINX_VERSION]-[STABILITY_TAG].
    image: wodby/drupal-nginx:8-1.13-3.0.1
#    image: wodby/drupal-nginx:7-1.13-3.0.1
#    image: wodby/drupal-nginx:6-1.13-3.0.1
#    image: wodby/drupal-nginx:8-1.12-3.0.1
#    image: wodby/drupal-nginx:7-1.12-3.0.1
#    image: wodby/drupal-nginx:6-1.12-3.0.1
    depends_on:
      - php
    environment:
      NGINX_STATIC_CONTENT_OPEN_FILE_CACHE: "off"
      NGINX_ERROR_LOG_LEVEL: debug
      NGINX_BACKEND_HOST: php
      NGINX_SERVER_ROOT: /var/www/html/web
    volumes:
      - codebase:/var/www/html
# Options for macOS users (https://docker4drupal.readthedocs.io/en/latest/macos)
#      - codebase:/var/www/html:cached # User-guided caching
#      - docker-sync:/var/www/html # Docker-sync
    labels:
      - 'traefik.backend=nginx'
      - 'traefik.port=80'
      - 'traefik.frontend.rule=Host:drupal.docker.localhost'

#  apache:
#    image: wodby/php-apache:2.4-2.0.0
#    depends_on:
#      - php
#    environment:
#      APACHE_LOG_LEVEL: debug
#      APACHE_BACKEND_HOST: php
#      APACHE_SERVER_ROOT: /var/www/html/web
#    volumes:
#      - codebase:/var/www/html
##      - docker-sync:/var/www/html # Docker-sync
#    labels:
#      - 'traefik.backend=apache'
#      - 'traefik.port=80'
#      - 'traefik.frontend.rule=Host:drupal.docker.localhost'

#  varnish:
#    image: wodby/drupal-varnish:4.1-2.2.0
#    depends_on:
#      - nginx
#    environment:
#      VARNISH_SECRET: secret
#      VARNISH_BACKEND_HOST: nginx
#      VARNISH_BACKEND_PORT: 80
#    labels:
#      - 'traefik.backend=varnish'
#      - 'traefik.port=6081'
#      - 'traefik.frontend.rule=Host:varnish.drupal.docker.localhost'

#  redis:
#    image: wodby/redis:3.2-2.1.3
##    image: wodby/redis:4.0-2.1.3

#  adminer:
#    image: wodby/adminer:4.3-1.1.0
#    environment:
#      ADMINER_SALT: adminer-salt
#    labels:
#      - 'traefik.backend=adminer'
#      - 'traefik.port=9000'
#      - 'traefik.frontend.rule=Host:adminer.drupal.docker.localhost'

#  pma:
#    image: phpmyadmin/phpmyadmin
#    environment:
#      PMA_HOST: mariadb
#      PMA_USER: drupal
#      PMA_PASSWORD: drupal
#      PHP_UPLOAD_MAX_FILESIZE: 1G
#      PHP_MAX_INPUT_VARS: 1G
#    labels:
#      - 'traefik.backend=pma'
#      - 'traefik.port=80'
#      - 'traefik.frontend.rule=Host:pma.drupal.docker.localhost'

#  solr:
## wodby/drupal-solr:[DRUPAL_VERSION]-[SOLR_VERSION]-[STABILITY_TAG].
#    image: wodby/drupal-solr:8-6.6-2.2.0
##    image: wodby/drupal-solr:7-5.4-2.2.0
#    environment:
#      SOLR_HEAP: 1024m
#    labels:
#      - 'traefik.backend=solr'
#      - 'traefik.port=8983'
#      - 'traefik.frontend.rule=Host:solr.drupal.docker.localhost'

  mailhog:
    image: mailhog/mailhog
    labels:
      - 'traefik.backend=mailhog'
      - 'traefik.port=8025'
      - 'traefik.frontend.rule=Host:mailhog.drupal.docker.localhost'

#  nodejs:
#    image: wodby/drupal-node:1.0-1.0.0
#    environment:
#       NODE_SERVICE_KEY: node-service-key
#    labels:
#      - 'traefik.backend=nodejs'
#      - 'traefik.port=8080'
#      - 'traefik.frontend.rule=Host:nodejs.drupal.docker.localhost'
#    volumes:
#      - ./path/to/your/single-page-app:/app
#    command: sh -c 'npm install && npm run start'

#  node:
#    image: node:alpine
#    working_dir: /app
#    labels:
#      - 'traefik.backend=node'
#      - 'traefik.port=3000'
#      - 'traefik.frontend.rule=Host:front.drupal.docker.localhost'
#    expose:
#      - "3000"
#    volumes:
#      - ./path/to/your/single-page-app:/app
#    command: sh -c 'npm install && npm run start'

#  memcached:
#    image: wodby/memcached:1.4-2.0.0

#  rsyslog:
#    image: wodby/rsyslog

#  athenapdf:
#    image: arachnysdocker/athenapdf-service:2.10.0
#    environment:
#      WEAVER_AUTH_KEY: weaver-auth-key
#      WEAVER_ATHENA_CMD: "athenapdf -S"
#      WEAVER_MAX_WORKERS: 10
#      WEAVER_MAX_CONVERSION_QUEUE: 50
#      WEAVER_WORKER_TIMEOUT: 90
#      WEAVER_CONVERSION_FALLBACK: false

#  blackfire:
#    image: blackfire/blackfire
#    environment:
#      BLACKFIRE_SERVER_ID: XXXXX
#      BLACKFIRE_SERVER_TOKEN: YYYYY

#  webgrind:
#    image: wodby/webgrind:1.5-1.0.0
#    environment:
#      WEBGRIND_PROFILER_DIR: /mnt/files/xdebug/profiler
#    labels:
#      - 'traefik.backend=webgrind'
#      - 'traefik.port=8080'
#      - 'traefik.frontend.rule=Host:webgrind.drupal.docker.localhost'
#    volumes:
#      - files:/mnt/files

  portainer:
    image: portainer/portainer
    command: --no-auth -H unix:///var/run/docker.sock
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    labels:
      - 'traefik.backend=portainer'
      - 'traefik.port=9000'
      - 'traefik.frontend.rule=Host:portainer.drupal.docker.localhost'

  traefik:
    image: traefik
    command: -c /dev/null --web --docker --logLevel=INFO
    ports:
      - '8000:80'
#      - '8080:8080' # Dashboard
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

volumes:
  codebase:
## Docker-sync for macOS users
#  docker-sync:
#    external: true
## For Xdebug profiler
#  files:

### Logs output
```
# Run "docker-compose logs [service]". Let's say you get 500 error for some reason then it'll be helpful to provide logs for php and http server (nginx/apache) services.


docker-compose logs php
```
php_1        | [08-Nov-2017 13:57:21] NOTICE: fpm is running, pid 1
php_1        | [08-Nov-2017 13:57:21] NOTICE: ready to handle connections
php_1        | 172.26.0.7 -  08/Nov/2017:13:57:30 +0000 "GET /index.php" 200
php_1        | [08-Nov-2017 13:57:30] WARNING: [pool www] child 51 said into stderr: "[08-Nov-2017 13:57:30 UTC] PHP Warning:  require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17"
php_1        | [08-Nov-2017 13:57:30] WARNING: [pool www] child 51 said into stderr: "[08-Nov-2017 13:57:30 UTC] PHP Fatal error:  require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17"
php_1        | [08-Nov-2017 13:57:31] WARNING: [pool www] child 50 said into stderr: "[08-Nov-2017 13:57:31 UTC] PHP Warning:  require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17"
php_1        | [08-Nov-2017 13:57:31] WARNING: [pool www] child 50 said into stderr: "[08-Nov-2017 13:57:31 UTC] PHP Fatal error:  require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17"
php_1        | 172.26.0.7 -  08/Nov/2017:13:57:31 +0000 "GET /index.php" 200
php_1        | [08-Nov-2017 13:57:33] WARNING: [pool www] child 51 said into stderr: "[08-Nov-2017 13:57:33 UTC] PHP Warning:  require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17"
php_1        | 172.26.0.7 -  08/Nov/2017:13:57:33 +0000 "GET /index.php" 200
php_1        | [08-Nov-2017 13:57:33] WARNING: [pool www] child 51 said into stderr: "[08-Nov-2017 13:57:33 UTC] PHP Fatal error:  require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17"
php_1        | [08-Nov-2017 13:57:40] WARNING: [pool www] child 50 said into stderr: "[08-Nov-2017 13:57:40 UTC] PHP Warning:  require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17"
php_1        | 172.26.0.7 -  08/Nov/2017:13:57:40 +0000 "GET /index.php" 200
php_1        | [08-Nov-2017 13:57:40] WARNING: [pool www] child 50 said into stderr: "[08-Nov-2017 13:57:40 UTC] PHP Fatal error:  require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17"
php_1        | [08-Nov-2017 13:57:44] WARNING: [pool www] child 51 said into stderr: "[08-Nov-2017 13:57:44 UTC] PHP Warning:  require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17"
php_1        | 172.26.0.7 -  08/Nov/2017:13:57:44 +0000 "GET /index.php" 200
php_1        | [08-Nov-2017 13:57:44] WARNING: [pool www] child 51 said into stderr: "[08-Nov-2017 13:57:44 UTC] PHP Fatal error:  require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17"
php_1        | 172.26.0.7 -  08/Nov/2017:13:57:46 +0000 "GET /index.php" 200
php_1        | [08-Nov-2017 13:57:46] WARNING: [pool www] child 50 said into stderr: "[08-Nov-2017 13:57:46 UTC] PHP Warning:  require(/var/www/html/web/../vendor/autoload.php): failed to open stream: No such file or directory in /var/www/html/web/autoload.php on line 17"
php_1        | [08-Nov-2017 13:57:46] WARNING: [pool www] child 50 said into stderr: "[08-Nov-2017 13:57:46 UTC] PHP Fatal error:  require(): Failed opening required '/var/www/html/web/../vendor/autoload.php' (include_path='.:/usr/local/lib/php') in /var/www/html/web/autoload.php on line 17"

**Comment by csandanov** [comment]
It seems composer just did not fetch your dependencies, make sure `composer update` properly installed drupal core and libs.

**Comment by shrop** [comment]
I ran into this issue and was able to resolve it by running the command below while accessing the php container via `docker exec -it <container name> /bin/sh`.

`rsync -rlt "/usr/src/drupal/" "${APP_ROOT}/"`

Found that command in https://github.com/wodby/drupal/blob/master/8/actions/init.sh. Looks like that command just doesn't populate all of the dependencies needed in `/var/www/html`. I was missing `composer.*` files and the `vendor` folder initially.

**Comment by FatGuyLaughing** [comment]
I get this on a fresh installation of everything. Have we found an automated solution, instead of having to manually rerun the command?

**Comment by csandanov** [comment]
So I guess this is the problem https://github.com/wodby/drupal/issues/6

**Comment by csandanov** [comment]
Resolved in [5.0.0](https://github.com/wodby/docker4drupal/releases/tag/5.0.0)
